routemaster/conftest.py

```diff
@@ -165,8 +235,44 @@ def create_deleted_label(create_label, delete_label):
     Create a label in the database and then delete it.
     """
 
-    def _create_and_delete(name: str, state_machine_name: str) -> None:
+    def _create_and_delete(name: str, state_machine_name: str) -> LabelRef:
         create_label(name, state_machine_name, {})
         delete_label(name, state_machine_name)
+        return LabelRef(name, state_machine_name)
 
     return _create_and_delete
+
+
+@pytest.fixture()
+def mock_webhook():
+    """Mock the test config's webhook call."""
+    @contextlib.contextmanager
+    def _mock(result=WebhookResult.SUCCESS):
+        runner = mock.Mock(return_value=result)
+        with mock.patch(
+            'routemaster.webhooks.RequestsWebhookRunner',
+            return_value=runner,
+        ):
+            yield runner
+    return _mock
+
+
+@pytest.fixture()
+def mock_test_feed():
+    """Mock out the test feed."""
+    @contextlib.contextmanager
+    def _mock(data={'should_do_alternate_action': False}):
+        httpretty.enable()
+        httpretty.register_uri(
+            httpretty.GET,
+            'http://localhost/tests',
+            body=json.dumps(data),
+            content_type='application/json',
+        )
+
+        yield
```

Comment:
Should we `try..finally` around this `yield` to ensure that we clean up the `httpretty` config?

Reply:
e6a226ca27e749c26b507f5d1bbc1ba5c811f9f8